validation-test/stdlib/SetAlgebra.swift

```diff
@@ -0,0 +1,181 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+import StdlibCollectionUnittest
+
+extension MinimalEquatableValue : Hashable {
+  public var hashValue: Int {
+    return value
+  }
+}
+
+public struct DefaultedSetAlgebra<Element : Hashable> : SetAlgebra {
+  var store = Set<Element>()
+  public init(){}
+  public init(_ elements: [Element]) {
+    for el in elements {
+      store.insert(el)
+    }
+  }
+
+  public func contains(_ member: Element) -> Bool {
+    return store.contains(member)
+  }
+
+  public func union(_ other: DefaultedSetAlgebra) -> DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.union(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public func intersection(_ other: DefaultedSetAlgebra) ->
+  DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.intersection(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public func symmetricDifference(_ other: DefaultedSetAlgebra) ->
+  DefaultedSetAlgebra {
+    var defaultedSetAlgebra = DefaultedSetAlgebra()
+    defaultedSetAlgebra.store = store.symmetricDifference(other.store)
+    return defaultedSetAlgebra
+  }
+
+  public mutating func insert(_ newMember: Element) -> (inserted: Bool,
+  memberAfterInsert: Element) {
+    return store.insert(newMember)
+  }
+
+  public mutating func remove(_ member: Element) -> Element? {
+    return store.remove(member)
+  }
+
+  public mutating func update(with newMember: Element) -> Element? {
+    return store.update(with: newMember)
+  }
+
+  public mutating func formUnion(_ other: DefaultedSetAlgebra) {
+    store.formUnion(other.store)
+  }
+
+  public mutating func formSymmetricDifference(_ other: DefaultedSetAlgebra) {
+    store.formSymmetricDifference(other.store)
+  }
+
+  public mutating func formIntersection(_ other: DefaultedSetAlgebra) {
+    store.formIntersection(other.store)
+  }
+}
+
+public func ==<Element>(lhs: DefaultedSetAlgebra<Element>,
+  rhs: DefaultedSetAlgebra<Element>) -> Bool {
+  return lhs.store == rhs.store
+}
+
+let SetAlgebraTests = TestSuite("SetAlgebra")
+
+SetAlgebraTests.test("contains") {
+  typealias P = DefaultedSetAlgebra<MinimalEquatableValue>
+  for test in findTests {
+    expectEqual(test.expected != nil,
+      P(test.sequence).contains(test.element))
+  }
+}
+
+SetAlgebraTests.test("union") {
+  for test in unionTests {
+    expectEqual(DefaultedSetAlgebra(test.expected),
```

Comment:
I recommend breaking the line after `(` if the function call spans multiple lines.  (Here and in other calls.)